run.sh

```diff
@@ -23,104 +23,103 @@ LOG_GROUP_NAME="/ccao/jobs/sqoop"
 echo "Total extraction time: ${HH}:${MM}:${SS} (hh:mm:ss)" \
     | ts '%.s' \
-    | tee -a ${TEMP_LOG_FILE}
+    | tee -a "$TEMP_LOG_FILE"
 
 # Bash function to convert text log to JSON consumable by CloudWatch
-parse_logs () {
-    cat $1 \
-    | sed 's/ /|/' \
-    | sed 's/\([0-9]\.[0-9]\{3\}\)[0-9]\{1,\}/\1/' \
-    | sed 's/\.//' \
-    | awk '{print $1, $NF}' OFS='|' FS='|' \
-    | jq -Rn '[inputs
+parse_logs() {
+    sed 's/ /|/' "$1" \
+        | sed 's/\([0-9]\.[0-9]\{3\}\)[0-9]\{1,\}/\1/' \
+        | sed 's/\.//' \
+        | awk '{print $1, $NF}' OFS='|' FS='|' \
+        | jq -Rn '[inputs
         | . / "\n"
         | (.[] | select(length > 0) | . / "|") as $input
         | {"timestamp": $input[0]|tonumber, "message": $input[1]}]' \
-    > $2
+            > "$2"
 }
 
 # Create log stream in CloudWatch with today's date
 LOG_STREAM_NAME="sqoop-job-log-$(date -u +'%Y-%m-%d_%H-%M-%S')"
 /usr/bin/aws logs create-log-stream \
-    --log-group-name ${LOG_GROUP_NAME} \
-    --log-stream-name ${LOG_STREAM_NAME}
+    --log-group-name "$LOG_GROUP_NAME" \
+    --log-stream-name "$LOG_STREAM_NAME"
 
 # Convert text output from Docker and AWS CLI to clean JSON
 # for upload to AWS CloudWatch. Logs need to be split into chunks for upload
 # First loop skips using sequence token necessary for subsequent uploads
 COUNTER=1
-split -l 4000 ${TEMP_LOG_FILE} logs/temp-sqoop-log-
+split -l 4000 "$TEMP_LOG_FILE" logs/temp-sqoop-log-
 for f in logs/temp-sqoop-log-*; do
-    parse_logs ${f} ${f}.json
-    if [[ ${COUNTER} -eq 1 ]]; then
+    parse_logs "$f" "$f".json
+    if [[ "$COUNTER" -eq 1 ]]; then
         SEQ_TOKEN=$(
             /usr/bin/aws logs put-log-events \
-                --log-group-name ${LOG_GROUP_NAME} \
-                --log-stream-name ${LOG_STREAM_NAME} \
-                --log-events file://${f}.json \
-            | jq -r .nextSequenceToken
+                --log-group-name "$LOG_GROUP_NAME" \
+                --log-stream-name "$LOG_STREAM_NAME" \
+                --log-events file://"$f".json \
+                | jq -r .nextSequenceToken
         )
     else
         SEQ_TOKEN=$(
             /usr/bin/aws logs put-log-events \
-                --log-group-name ${LOG_GROUP_NAME} \
-                --log-stream-name ${LOG_STREAM_NAME} \
-                --log-events file://${f}.json \
-                --sequence-token ${SEQ_TOKEN} \
-            | jq -r .nextSequenceToken
+                --log-group-name "$LOG_GROUP_NAME" \
+                --log-stream-name "$LOG_STREAM_NAME" \
+                --log-events file://"$f".json \
+                --sequence-token "$SEQ_TOKEN" \
+                | jq -r .nextSequenceToken
         )
     fi
     COUNTER=$((COUNTER + 1))
 done
 echo "Logs successfully uploaded to CloudWatch"
 
 # Remove uploaded log files
-mv ${TEMP_LOG_FILE} ${BACKUP_LOG_FILE}
-rm ${TEMP_LOG_FILE}*
+mv ./"$TEMP_LOG_FILE" ./"$BACKUP_LOG_FILE"
+if [[ -n "$TEMP_LOG_FILE" ]]; then rm ./"$TEMP_LOG_FILE"*; fi
```

Comment:
Changing this mega dum-dum line so that `rm *` and variants cannot run.

Reply:
**[Question, non-blocking]** I don't totally get what this line is for, in what case would `-n "$TEMP_LOG_FILE` evaluate as false? Isn't `TEMP_LOG_FILE` always set to a non-empty string on line 5? I think I'm probably just misunderstanding this!

Reply:
Basically only for the case in which someone is playing around with the script and deletes that line, then re-runs the script. Saving us from our future selves.